src/west/commands/build.py

```diff
@@ -188,6 +189,20 @@ def _setup_source_dir(self):
             source_dir = os.getcwd()
         self.source_dir = os.path.abspath(source_dir)
 
+    def _get_available_boards_str(self):
+        path = Path(os.getenv("ZEPHYR_BASE")+"/boards")
+        brds_paths = path.glob("**/Kconfig.board")
```

Comment:
I think you should take a look at usage.cmake in the Zephyr repository to see how they do it. Maybe you can use the same approach here, or even better, try to extract that component as a script you can run with `cmake -P`, so there is just one implementation. Either way will keep west consistent with cmake (the latter way of running cmake directly is preferred of course).